socket_mode_client.go

```diff
@@ -0,0 +1,115 @@
+package slack
+
+import (
+	"encoding/json"
+	"net/url"
+	"sync"
+	"time"
+
+	"github.com/gorilla/websocket"
+)
+
+type SocketModeConnectedEvent struct {
+	ConnectionCount int // 1 = first time, 2 = second time
+	Info            *SocketModeConnection
+}
+
+// SocketModeConnection contains various details about the SocketMode connection.
+// It is returned by an "apps.connections.open" API call.
+type SocketModeConnection struct {
+	URL  string                 `json:"url,omitempty"`
+	Data map[string]interface{} `json:"-"`
+}
+
+// SocketModeMessage maps to each message received via the WebSocket connection of SocketMode
+type SocketModeMessage struct {
+	Type       string                   `json:"type"`
+	Reason     string                   `json:"reason"`
+	Payload    SocketModeMessagePayload `json:"payload"`
+	EnvelopeID string                   `json:"envelope_id"`
```

Comment:
FYI: https://github.com/slackapi/python-slack-sdk/blob/v3.2.0/slack_sdk/socket_mode/request.py#L7-L12

A Socket Mode envelope can have the following attributes.

* AcceptsResponsePayload bool
* RetryAttempt *int // only for Events API
* RetryReason string // only for Events API

Reply:
Thanks! I've added a bunch of fields according to your info and my observation

https://github.com/mumoshu/slack/blob/852061eaff58f9acb2a18f70ac871341d0f57982/socketmode/request.go#L15-L38